tests/zfs-tests/tests/functional/snapshot/snapshot_018_pos.ksh

```diff
@@ -0,0 +1,127 @@
+#! /bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2022 iXsystems, Inc.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/snapshot/snapshot.cfg
+
+#
+# DESCRIPTION:
+# Verify the functionality of snap_cmtime property
+#
+# STRATEGY:
+# 1. Create a pool
+# 2. Verify snap_cmtime property is NULL
+# 3. Create a filesystem
+# 4. Verify snap_cmtime property is NULL
+# 5. Create snapshots for all filesystems
+# 6. Verify snap_cmtime property shows correct time
+# 7. Unmount all filesystems
+# 8. Create a snapshot while unmounted
+# 9. Verify snap_cmtime
+# 10. Mount the filsystems
+# 11. Verify snap_cmtime
+# 12. Destroy the snapshots
+# 13. Verify snap_cmtime
+#
+
+function cleanup
+{
+	datasetexists $TESTPOOL/$TESTFS && destroy_dataset $TESTPOOL/$TESTFS -R
```

Comment:
Extra snapshots are potentially left on the pool if something in the middle of the test fails. That should be cleaned up as well. Maybe we should just recreate the pool from scratch at this point.

Reply:
Thank you @freqlabs. I have recreated the pool from scratch.